src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/ReflectionFunction.cs

```diff
@@ -270,9 +277,38 @@ public async Task<FormulaValue> InvokeAsync(IServiceProvider serviceProvider, Fo
 
             formulaResult ??= FormulaValue.NewBlank(_info.RetType);
 
-            if (!formulaResult.Type._type.Accepts(_info.RetType._type, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true))
+            var dataSourceType = formulaResult.Type._type;
+            var retType = _info.RetType._type;
+            
+            if (!dataSourceType.Accepts(retType, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true))
             {
-                return CommonErrors.CustomError(formulaResult.IRContext, string.Format(CultureInfo.InvariantCulture, "Return type should have been {0}, found {1}", _info.RetType._type, formulaResult.Type._type));
+                bool isVaid = false;
+
+                if (retType.IsRecord)

```

Comment:
>IsRecord [](http://example.com/codeflow?start=28&length=8)

Similar comment than the one in CustomTexlFunction - if we are checking for records we should probably check for tables as well

Reply:
update to check table as well.